src/Aspire.Hosting.Azure.Functions/AzureFunctionsProjectResourceExtensions.cs

```diff
@@ -116,13 +116,9 @@ public static IResourceBuilder<AzureFunctionsProjectResource> WithHostStorage(th
     public static IResourceBuilder<AzureFunctionsProjectResource> WithReference<TSource>(this IResourceBuilder<AzureFunctionsProjectResource> destination, IResourceBuilder<TSource> source, string? connectionName = null)
         where TSource : IResourceWithConnectionString, IResourceWithAzureFunctionsConfig
     {
-        // REVIEW: There's a conflict with the connection strings formats and various azure functions extensions
```

Comment:
I think a comment (maybe even in `<remarks`) might still be waranted about what we are doing here and why.

I think people will be suprised to see that the connection information "moved" on them when they are in a functions app. For example, if my function needs to talk to a database, I might need to read the connection string myself and do something with it (outside of an Aspire client library).

Reply:
ACK -- I'll add a note here.